private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/field_default.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufbreakingcheck
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/bufbuild/buf/private/pkg/slicesext"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+type fieldDefault struct {
+	comparable any
```

Comment:
```suggestion
	comparable comparable
```
Will need to update `getDefault` to return an error if it can't type-assert the value to `comparable`, too.

Reply:
@bufdev, doh, this isn't possible. The `comparable` type uses a "type term", which means it can only be used as a constraint for a type parameter, and not as a type for a variable, parameter, or struct element. And I can't make this generic (and use a type parameter) because the types aren't known at compile-time: the value is coming from an `any` (returned by `protoreflect.Value.Interface()`).